datalad/interface/add_archive_content.py

```diff
@@ -186,238 +207,316 @@ class AddArchiveContent(Interface):
         delete_after=Parameter(
             args=("--delete-after",),
             action="store_true",
-            doc="""extract under a temporary directory, git-annex add, and delete after.  To
-             be used to "index" files within annex without actually creating corresponding
-             files under git.  Note that `annex dropunused` would later remove that load"""),
+            doc="""extract under a temporary directory, git-annex add, and
+            delete afterwards. To be used to "index" files within annex without
+            actually creating corresponding files under git. Note that
+            `annex dropunused` would later remove that load"""),
 
         # TODO: interaction with archives cache whenever we make it persistent across runs
         archive=Parameter(
             doc="archive file or a key (if %s specified)" % _KEY_OPT,
             constraints=EnsureStr()),
     )
 
-        # use-case from openfmri pipeline
-        #     ExtractArchives(
-        #         # will do the merge of 'replace' strategy
-        #         source_branch="incoming",
-        #         regex="\.(tgz|tar\..*)$",
-        #         renames=[
-        #             ("^[^/]*/(.*)", "\1") # e.g. to strip leading dir, or could prepend etc
-        #         ],
-        #         #exclude="license.*",  # regexp
-        #     ),
-
     @staticmethod
-    def __call__(archive, annex=None,
-                 add_archive_leading_dir=False,
-                 strip_leading_dirs=False, leading_dirs_depth=None, leading_dirs_consider=None,
-                 use_current_dir=False,
-                 delete=False, key=False, exclude=None, rename=None, existing='fail',
-                 annex_options=None, copy=False, commit=True, allow_dirty=False,
-                 stats=None, drop_after=False, delete_after=False):
-        """
-        Returns
-        -------
-        annex
-        """
+    @datasetmethod(name='add_archive_content')
+    @eval_results
+    def __call__(
+            archive,
+            dataset=None,
+            annex=None,
+            add_archive_leading_dir=False,
+            strip_leading_dirs=False,
+            leading_dirs_depth=None,
+            leading_dirs_consider=None,
+            use_current_dir=False,
+            delete=False,
+            key=False,
+            exclude=None,
+            rename=None,
+            existing='fail',
+            annex_options=None,
+            copy=False,
+            commit=True,
+            allow_dirty=False,
+            stats=None,
+            drop_after=False,
+            delete_after=False):
+
```

Comment:
Needs `if annex is not None: raise ...DeprecationsWarning`, and possibly setting `dataset` accordingly, if not set already (if so, raise `ValueError`), for a transition period.

Reply:
OK, found below. Could move up here.